velox/functions/sparksql/specialforms/DecimalRound.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/specialforms/DecimalRound.h"
+#include "velox/expression/ConstantExpr.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+template <typename R, typename A>
+class DecimalRoundFunction : public exec::VectorFunction {
+ public:
+  DecimalRoundFunction(
+      int32_t scale,
+      uint8_t inputPrecision,
+      uint8_t inputScale,
+      uint8_t resultPrecision,
+      uint8_t resultScale)
+      : scale_(scale),
+        inputPrecision_(inputPrecision),
+        inputScale_(inputScale),
+        resultPrecision_(resultPrecision),
+        resultScale_(resultScale) {
+    const auto [p, s] = DecimalRoundCallToSpecialForm::getResultPrecisionScale(
+        inputPrecision, inputScale, scale);
+    VELOX_USER_CHECK_EQ(
+        p,
+        resultPrecision,
+        "The result precision of decimal_round should be the same with expected.");
```

Comment:
> should be the same with expected

This is not very informative, is it? Would it be possible to rewrite the error message for clarity?